### PR TITLE
git: ignore IDE folders and build arterfacts

### PR DESCRIPTION
**Pull request by maximilien-noal**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,23 @@
 .DS_Store
 /target
+
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
```
Comment (madmoose) [inline .gitignore]: These two seem redundant with `/target` above. Can you remove them and put the comment above `/target`? Actually, put `/target/`,

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```
Comment (madmoose) [inline .gitignore]: We're creating an executable, so Cargo.lock should *not* be in .gitignore. Please remove.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
+
+# VSCode
+.vscode/
```
Comment (madmoose) [inline .gitignore]: Please use `/.vscode/` for directories in the project root.

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,23 @@
...
+
+# JetBrains Rider / IDEA
+.idea/
```
Comment (madmoose) [inline .gitignore]: Same here, `/.idea/`.

**Review by madmoose — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (maximilien-noal) [reply .gitignore]: Done.

File: .gitignore (hunk 3 above)
Comment (maximilien-noal) [reply .gitignore]: Done.

File: .gitignore (hunk 2 above)
Comment (maximilien-noal) [reply .gitignore]: Done.

File: .gitignore (hunk 1 above)
Comment (maximilien-noal) [reply .gitignore]: Done.

**Pull request merged**